junit4/src/test/java/org/jboss/weld/junit4/bean/AlternativeMockBeanTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2017, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.weld.junit4.bean;
+
+import static org.junit.Assert.assertEquals;
+
+import javax.enterprise.context.Dependent;
+
+import org.jboss.weld.junit.MockBean;
+import org.jboss.weld.junit4.WeldInitiator;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ *
+ * @author Martin Kouba
+ */
+public class AlternativeMockBeanTest {
+
+    @Rule
+    public WeldInitiator weld = WeldInitiator.from(SimpleService.class)
+            .addBeans(MockBean.builder().types(MyService.class).selectedAlternative().beanClass(CoolService.class).create(c -> new CoolService()).build())
```

Comment:
How about merging `selectedAlternative()` and `beanClass()` into one method? Since you need to specify `beanClass()` when selecting alternative, it might as well be a parameter of the `selectedAlternative()` method, right?
Such as `public Builder<T> selectedAlternative(Class<?> beanClass)`.

Reply:
I don't like it much. First, in some cases it might be useful _not_ to specify the bean class (e.g. if you want to select all mock alternative beans by default). Also I'm not sold on the idea of combining too many things in one "combo" method - I think that it's enough that we do `alternative(true)`...

Reply:
Then maybe having both methods would make sense? E.g. either just `selectedAlternative()` or `selectedAlternative(Class<?> clazz)`

Reply:
+1